clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2231,13 +2231,14 @@ llvm::Constant *CodeGenModule::EmitAnnotateAttr(llvm::GlobalValue *GV,
                  *UnitGV = EmitAnnotationUnit(L),
                  *LineNoCst = EmitAnnotationLineNo(L);
 
+  llvm::Type *ResType = llvm::PointerType::getInt8PtrTy(
+      this->getLLVMContext(), GV->getType()->getPointerAddressSpace());
+  llvm::Constant *C =
+      llvm::ConstantExpr::getPointerBitCastOrAddrSpaceCast(GV, ResType);
   // Create the ConstantStruct for the global annotation.
   llvm::Constant *Fields[4] = {
-    llvm::ConstantExpr::getBitCast(GV, Int8PtrTy),
-    llvm::ConstantExpr::getBitCast(AnnoGV, Int8PtrTy),
-    llvm::ConstantExpr::getBitCast(UnitGV, Int8PtrTy),
-    LineNoCst
-  };
+      C, llvm::ConstantExpr::getBitCast(AnnoGV, Int8PtrTy),
```

Comment:
How about using `getPointerBitCastOrAddrSpaceCast` here directly? Also, I think it is more readable when each element is placed on its own line (difficult to parse otherwise, since they are pretty long).

Reply:
Sure, I can use this cast directly.
What about readability - I agree, but this formatting is done by clang-format and it's preferable to use this code-style

Reply:
I think that if you can format a code better than clang-format without violating the coding style rules, then you should definitely do this.